javascript/ql/test/library-tests/frameworks/Angular2/sink.component.ts

```diff
@@ -1,22 +1,22 @@
-import { Component } from "@angular/core";
+import { Component, Input } from "@angular/core";
 import { DomSanitizer } from '@angular/platform-browser';
 
 @Component({
     selector: "sink-component",
     template: "not important"
 })
 export class SinkComponent {
-    sink1: string;
-    sink2: string;
-    sink3: string;
-    sink4: string;
-    sink5: string;
-    sink6: string;
-    sink7: string;
-    sink8: string;
-    sink9: string;
+    @Input() sink1: string;
+    @Input() sink2: string;
+    @Input() sink3: string;
+    @Input() sink4: string;
+    @Input() sink5: string;
+    @Input() sink6: string;
+    @Input() sink7: string;
+    @Input() sink8: string;
+    @Input() sink9: string;
```

Comment:
all of these are named "sink", but they're sources, right?

Reply:
They're like React props, they're just passed in from the instantiation site. When I initially wrote this test long ago, I had omitted the `@Input()` decorator on these fields, which you're supposed to put on such input fields in Angular (not sure if they're strictly required though).

If the threat model is enabled they become sources (in addition to having incoming flow from instantiation sites). The unit test now reports which nodes are associated with this threat model, though the data flow test doesn't treat them as taint sources.